cloudinit/config/cc_lxd.py

```diff
@@ -281,16 +281,25 @@ def maybe_cleanup_default(net_name, did_init, create, attach,
         return
 
     fail_assume_enoent = "failed. Assuming it did not exist."
+    failed = "failed."
     succeeded = "succeeded."
     if create:
-        msg = "Deletion of lxd network '%s' %s"
+        msg = "Detach of lxd network '%s' from profile '%s' %s"
         try:
-            _lxc(["network", "delete", net_name])
-            LOG.debug(msg, net_name, succeeded)
+            _lxc(["network", "detach-profile", net_name, profile])
```

Comment:
do lxc2 and lxc3 have detach-profile ?
if not, then this will fail for reason other than "no bridge", and we will not attempt to delete the network.

Reply:
Oh well. `detach-profile` was introduced together with `lxc network`, so I assumed it was there in lxc2 too, but lxc2 doesn't have `lxc network` at all...

Reply:
@smoser so basically `lxc detach-profile ...` will fail in Xenial in the exact same way `lxc network delete ...` used to fail, that is:
 - fail with RC=1
 - RC=1 will cause cloud-init follow the fail_assume_enoent path
In other words we're going to treat LXD2 (= Xenial) exactly as before.